doc/source/dev/plugins.rst

```diff
@@ -0,0 +1,88 @@
+.. devguide_plugins:
+
+Developing Plugins
+------------------
+
+As discussed in the user's guide, WATTS has a variety of plugins for different
+codes (primarily focused on application codes in nuclear science and
+engineering) that are already available. If you wish to use WATTS with a code
+that does not already have a plugin available, extending WATTS to handle another
+code can be done quite easily.
+
+There are two ways to incorporate new codes into WATTS. First, if the code uses
+a text-based input, the :class:`~watts.PluginTemplate` class can be used to
+specify the executable, how it should be invoked from a command-line, and a
+template for the input file. For example, if you have a code called ``goblin``
+that should be executed from a command line as:
+
+.. code-block:: sh
+
+    goblin input=filein.gbl
+
+a plugin can be set up as::
+
+    goblin = watts.PluginTemplate(
+        executable='goblin',
+        execute_command=['{self.executable}', 'input={self.input_name}'],
+        template_file='filein.gbl'
+    )
```

Comment:
this looks awesome! - but I am not clear what `execute_command` structure is. why do you need a list? - maybe it should just take a string like this: '{self.executable} input={self.input_name}' that is customizable.

Reply:
Each item in the list is a separate command-line argument. It's easy enough for me to modify this to accept either of list of strings or just a single string that represents the whole command line (in which case it will get converted to a list internally).